plugins/commands/snapshot/command/push.rb

```diff
@@ -0,0 +1,33 @@
+require 'json'
+require 'optparse'
+
+require_relative "push_shared"
+
+module VagrantPlugins
+  module CommandSnapshot
+    module Command
+      class Push < Vagrant.plugin("2", :command)
+        include PushShared
+
+        def execute
+          opts = OptionParser.new do |o|
+            o.banner = "Usage: vagrant snapshot push [options] [vm-name]"
+            o.separator ""
+            o.separator "Take a snapshot of the current state of the machine and 'push'"
```

Comment:
"of the given machine"

I assume this is going to behave like most other Vagrant commands - in a single-machine Vagrantfile, it will not require the name argument, but in a multi-machine Vagrantfile, you'll need to specify it?